libraries/core/src/main/kotlin/com/bumble/appyx/core/children/ChildAwareImpl.kt

```diff
@@ -53,7 +53,7 @@ class ChildAwareImpl<N : Node> : ChildAware<N> {
             }
     }
 
-    override fun <T : Node> whenChildAttached(
+    override fun <T : Any> whenChildAttached(
         child: KClass<T>,
         callback: ChildCallback<T>
     ) {
```

Comment:
Maybe check if `child` is a `Node` to fail early? Rather that never get a callback?

Reply:
But it is not, that the main point. In my sample `Child` is just an interface, like we have in our templates.